enhancements/ingress/customize-error-code-pages.md

```diff
@@ -0,0 +1,121 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+## Summary
+There is no supported and reasonable method to customize the HAProxy 2.x error page as deployed in OpenShift.
+Customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711).
+does not work with OCP 4. It's also important that the right page as per user requirements is served at the right moment.
+Example - 503 error page shall be returned if no pod is available. 
+But when the URI is wrong/does not exist, we should return 404 or a customized 404 page as per user requirements.
+## Motivation
+In OCP 3.11 customizing error pages is possible via a [documented procedure](https://access.redhat.com/solutions/3425711)
+Customers resist customizing the HAProxy error page per our recommended procedure (documented here: https://access.redhat.com/solutions/3425711)
+Also this procedure is not possible in OCP 4.
+So while it is possible to define custom Error Pages, it's also important that the right page is served at the right moment. Currently 503 is 
+returned if no pod is available and also when the URI has a wrong target. But when the URI is wrong/does not
+exist, we should return 404 or something different as it otherwise may confuse end-customers.
+
+### Goals
+1. Enable the cluster administrator to specify custom error code pages for haproxy router to serve them at the right moment.
+
+### Non-Goal
+TBD
+
+## Proposal
+0. User creates a custom error code page configmap `http503page` in `openshift-config`
+1. The IngressController API is extended by adding an optional
+`HttpErrorCodePage` field with type string
+```go
+//This used to sync the configmap having customizable error code page created by cluster admin in openshift-config namespace.
+//to openshift-ingress which will be used by the router to serve appropriate error code pages.
+//If this field is empty, the default values i.e default shipped error code pages will be served by the haproxy router.
+HttpErrorCodePage string `json:"httpErrorCodePage,omitempty"`
+
+```
+Using the default, the router uses the default error pages used by haproxy.
+The `HttpErrorCodePage` field enables the cluster administrator to
+specify custom error code pages.
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  httpErrorCodePage: "http503page"  // http503page is the config map name with custom 
+                                    //503 error page created in openshift-config by cluster-admin. 
+```
+2. `errorfile` stanzas need to be added as per https://www.haproxy.com/blog/serve-dynamic-custom-error-pages-with-haproxy/
+3. default error pages will be added in openshift/router for the haproxy.
+4. A  new controller to sync the cluster admin created configmap having custom error pages from `openshift-config` 
```

Comment:
Is syncing the config map from the `openshift-config` namespace to the `openshift-ingress` namespace necessary? Is there a reason we can't just use the config map in the original namespace directly?

Reply:
Ah I see, thanks for the historical explanation. Thanks!